pkg/command/command.go

```diff
@@ -0,0 +1,54 @@
+package command
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+)
+
+type Command struct {
```

Comment:
Keeping it similar to `command.Simple`, call this one `Compound` or `Complex`.

Reply:
I think Line is too vauge.

Reply:
Cool, I'll call it `Compound` then

Reply:
Also, I do not understand how this struct will be used. I dont think there is any situation where we will execute a list of simple commands, given that this is not a part of the parser.

Reply:
This struct is more for AST that will satisfy the Node interface that I'm thinking of designing. Consider the current bash shell, we have a simple command that gets executed but there are multiple commands as well that are piped together. These multiple commands get executed in a different order depending on the pipes and redirection. That's why this is a separate struct. I was reading [this](https://www.cs.purdue.edu/homes/grr/SystemsProgrammingBook/Book/Chapter5-WritingYourOwnShell.pdf) and considering this type of structure with a bit of modern take on it. Let me know if you have a different approach in mind.